packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseServerObject-test.js

```diff
@@ -0,0 +1,79 @@
+const { Fury } = require('fury');
+const { expect } = require('../../chai');
+
+const parse = require('../../../../lib/parser/oas/parseServerObject');
+const Context = require('../../../../lib/context');
+
+const { minim: namespace } = new Fury();
+
+describe('#parseHostsObject', () => {
+  let context;
+  beforeEach(() => {
+    context = new Context(namespace);
+  });
+
+  it('provides warning when server is non-object', () => {
+    const server = new namespace.elements.String();
+
+    const parseResult = parse(context)(server);
+
+    expect(parseResult.length).to.equal(1);
+    expect(parseResult).to.contain.warning("'Server Object' is not an object");
+  });
+
+  describe('#url', () => {
+    it('parse server object with URL', () => {
+      const server = new namespace.elements.Object({
+        url: 'https://{username}.gigantic-server.com/{version}',
+      });
+
+      const parseResult = parse(context)(server);
+      expect(parseResult).to.not.contain.annotations;
+      const resource = parseResult.get(0);
+      expect(resource).to.be.instanceof(namespace.elements.Resource);
+
+      const hostClass = resource.classes.getValue(0);
+      expect(hostClass).to.be.equal('host');
+
+      const href = resource.href.toValue();
+      expect(href).to.be.equal('https://{username}.gigantic-server.com/{version}');
+    });
+
+    it('warns when server object does not contain URL', () => {
+      const server = new namespace.elements.Object({
+      });
+
+      const parseResult = parse(context)(server);
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult).to.contain.warning("'Server Object' is missing required property 'url'");
+    });
```

Comment:
I think we're missing a test case for testing what happens when `url` is not a string. This is similiar to one of the below tests such as `warns when description is not a string`.

Reply:
all right I will do that